R/input-utils.R

```diff
@@ -90,31 +90,39 @@ generateOptions <- function(inputId, selected, inline, type = 'checkbox',
   div(class = "shiny-options-group", options)
 }
 
+# If a list/vector is unnamed, give it blank names
+ensureNamed <- function(x) {
+  if (is.null(names(x))) names(x) <- character(length(x))
+  x
+}
+
+# Take a vector or list, and convert to list. Also, if any children are
+# vectors with length > 1, convert those to list. If the list is unnamed,
+# convert it to a named list with blank names.
+listify <- function(x) UseMethod("listify", x)
+
+#' @export
+listify.list <- function(x) ensureNamed(lapply(x, listify))
+#' @export
+listify.NULL <- listify.list
+
+#' @export
+listify.character <- function(x) {
+  if (length(x) == 1 && is.null(names(x))) x else as.list(ensureNamed(x))
```

Comment:
`... else ensureNames(as.list(x))` ?

Is the goal to return a list with every element's name as `''` or to return a list with no names?

Reply:
In that case, return a list with every element's name as `''` (unless it's already named)

Reply:
That's my interpretation of https://github.com/rstudio/shiny/pull/2524/files#diff-c851d180beb8024d6b46a73dab4756d9L113 .  `makeNamed(as.list(val))` vs `as.list(ensureNamed(x))`

I like your version better.  "Set all the names to the character values if names does not exist"